### PR TITLE
CORE-8386 Add consul config template and update Dockerfile for it.

### PR DESCRIPTION
**Pull request by psarando**
This PR adds a consul config template that uses the new config values added by cyverse-de/ansible#40.

### CODE REVIEW

File: data-info.properties.tmpl [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{{ with $v := (key (printf "%s/irods/base" $base))        }}data-info.irods.home        = {{ $v }}{{ end }}
+{{ with $v := (key (printf "%s/irods/zone" $base))        }}data-info.irods.zone        = {{ $v }}{{ end }}
+{{ with $v := (key (printf "%s/irods/resc" $base))        }}data-info.irods.resc        = {{ $v }}{{ end }}
+{{ with $v := (key (printf "%s/irods/max-retries" $base)) }}data-info.irods.max-retries = {{ $v }}{{ end }}
+{{ with $v := (key (printf "%s/irods/retry-sleep" $base)) }}data-info.irods.retry-sleep = {{ $v }}{{ end }}
+{{ with $v := (key (printf "%s/irods/use-trash" $base))   }}data-info.irods.use-trash   = {{ $v }}{{ end }}
+{{ with $v := (key (printf "%s/irods/admin-users" $base)) }}data-info.irods.admin-users = {{ $v }}{{ end }}
+
+# file typing configuration
+data-info.type-detect.type-attribute = {{ (key (printf "%s/data-info/type-detect-attribute" $base)) }}
```
Comment (ianmcorvidae) [inline data-info.properties.tmpl]: I guess this one's required but the rest have defaults? Just making sure it's different for a reason, heh

Comment (psarando) [inline data-info.properties.tmpl]: Actually, there are other required properties besides this one, but they are interleaved with optional properties, and it was messing with the template's readability. So I just used the `with` syntax in those blocks and left it up to the service to complain if a property is missing.

I think the service will throw the same error if the `data-info.anon-files-base-url` property has a blank value or if it's not included at all.

Comment (psarando) [inline data-info.properties.tmpl]: Oh, and this line was already long enough without the `with` syntax 😄

**Comment by psarando** [comment]
Thanks for the review 👍

**Pull request merged**